### ISSUE TITLE
 loss = CEL_weighted(pred_scales, target_scales) 出现IndexError: Target -1 is out of bounds.越界。请问你知道是哪里有error吗

### ISSUE THREAD

**Issue by zhouqi946** [JindongJiang/RedNet#13]

**Comment by Jinczhg** [comment]
You might want to tweak the batch size, say lowering it down.

**Comment by Jinczhg** [comment]
有两种方式可以尝试：
1. 降低你的 pytorch 版本，比如1.2 及以下；
2. 在 crossentrophyloss 定义的那里，把
`mask = target > 0`
改成
`mask = target > 2e-9`

我debug的时候发现在把 label value: float64 0 读进 tensor 的时候，新版的 pytorch 有时候会把 0 当成很小的值对待比如 2e-9，这样在接下来 target - 1 然后 cast to long 的步骤里那些原本的 0 值就会变成 -1，从而报错。低版本的pytorch 好像没有遇到这种问题。

**Comment by klammecr** [comment]
I am using PyTorch 1.13 and Pytorch 2.0. This worked for me starting at line 50:

```
            targets = targets.long()
            mask = targets > 0
            targets_m = targets.clone()
```
